airflow/www/static/connection_form.js

```diff
@@ -7,7 +7,12 @@
         jdbc: {
             hidden_fields: ['port', 'schema', 'extra'],
             relabeling: {'host': 'Connection URL'},
+        },
+        google_cloud_platform: {
+            hidden_fields: ['host', 'schema', 'login', 'password', 'port', 'extra'],
```

Comment:
No, new connections should be set up using "project", "key_path", etc. fields that are displayed in the Airflow UI, not extras. I think "extras" is opaque for users who haven't read the code. However, if someone uses a connection that predates this change and DOES have an "extra" field, the code [here](https://github.com/airbnb/airflow/pull/1119/files#diff-a9d313c95097c265b6b934953bdfb2dbR78) will recognize that and handle it properly.

As far as I know, the connection type is largely irrelevant to Airflow -- you could create an "FTP" connection and pass it to `GCSHook`, and as long as it had the correct attributes it would work properly. The only thing that the type does affect is the fields that are displayed/handled by the UI (which are put in the "extras" field anyway behind the scenes)

So bottom line, lets say you have a connection of type `google_cloud_storage` with `key_path` in extras, and you've been using it with `GoogleCloudStorageHook`, you should be able to pass that same connection to a `GCSHook` no problem.